models/quality_measures/intermediate/cqm438_statin_therapy_for_patients_with_cardiovascular_disease/quality_measures__int_cqm438_denominator.sql

```diff
@@ -0,0 +1,282 @@
+{{ config(
+     enabled = var('quality_measures_enabled',var('claims_enabled',var('clinical_enabled',var('tuva_marts_enabled',False)))) | as_bool
+   )
+}}
+
+with patients_with_ascvd as (
+
+    select 
+          patient_id
+        , performance_period_begin
+        , performance_period_end
+        , measure_id
+        , measure_name
+        , measure_version
+        , 1 as criteria
+    from {{ ref('quality_measures__int_cqm438_denominator_criteria1') }}
+
+)
+
+, patients_with_cholesterol as (
+
+    select 
+          patient_id
+        , performance_period_begin
+        , performance_period_end
+        , measure_id
+        , measure_name
+        , measure_version
+        , 2 as criteria
+    from {{ ref('quality_measures__int_cqm438_denominator_criteria2') }}
+
+)
+
+, patients_with_diabetes as (
+
+    select 
+          patient_id
+        , performance_period_begin
+        , performance_period_end
+        , measure_id
+        , measure_name
+        , measure_version
+        , 3 as criteria
+    from {{ ref('quality_measures__int_cqm438_denominator_criteria3') }}
+
+)
+
+, visit_codes as (
+
+    select
+          code
+        , code_system
+    from {{ ref('quality_measures__value_sets') }} 
+    where lower(concept_name) in (
+          'annual wellness visit'
+        , 'office visit'
+        , 'outpatient consultation'
+        , 'preventive care services established office visit, 18 and up'
+        , 'preventive care services individual counseling'
+        , 'preventive care services initial office visit, 18 and up'
+    )
+
+)
+
+, visits_encounters as (
+
+    select patient_id
+         , coalesce(encounter.encounter_start_date,encounter.encounter_end_date) as min_date
+         , coalesce(encounter.encounter_end_date,encounter.encounter_start_date) as max_date
+    from {{ ref('quality_measures__stg_core__encounter') }} encounter
+    inner join {{ ref('quality_measures__int_cqm438__performance_period') }} as pp
+        on coalesce(encounter.encounter_end_date,encounter.encounter_start_date) >= pp.performance_period_begin
+        and  coalesce(encounter.encounter_start_date,encounter.encounter_end_date) <= pp.performance_period_end
+    where lower(encounter_type) in (
+          'home health'
+        , 'office visit'
+        , 'outpatient'
+        , 'outpatient rehabilitation'
+        , 'telehealth'
+        , 'nonacute inpatient'
+        , 'encounter inpatient'
+        , 'acute inpatient'
+        , 'emergency department'
```

Comment:
They are not concepts exactly but I was under the impression that they are defined with domain knowledge on what type of encounters to include.
For now, I have updated `encounter_type` with reference to [Breast Cancer Measure](https://github.com/tuva-health/tuva/blob/main/models/quality_measures/intermediate/nqf2372_breast_cancer_screening/quality_measures__int_nqf2372_denominator.sql).
Please let me know what you think.

Reply:
@utsavpaudel Gotcha, that makes sense to me since these do align with concepts from the measure itself. I just want to be careful we're not including encounter types that do not match concepts in the measure.